tests/e2e/root_v3_test.go

```diff
@@ -0,0 +1,45 @@
+package e2e_test
+
+import (
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+type RootV3Resource struct {
+	Links struct {
+		Self struct {
+			Href string `json:"href"`
+		} `json:"self"`
+	} `json:"links"`
+}
+
+var _ = Describe("TrailingSlash", func() {
+	var (
+		httpRespWithSlash    *resty.Response
+		httpRespWithoutSlash *resty.Response
+		resultWithoutSlash   RootV3Resource
+		resultWithSlash      RootV3Resource
+	)
+
+	When("Sending request to /v3 and /v3/", func() {
```

Comment:
As a confirmation to the test restructuring recommendation above, this `When` says `Sending request to /v3 and /v3/`. The fact that there is an `and` in the description tells that these are actually two contexts. Putting them in a single `When` sounds like a smell to me.

Reply:
Done